src/ucp/core/ucp_mm.c

```diff
@@ -1399,32 +1399,38 @@ static ucs_rcache_ops_t ucp_mem_rcache_ops = {
 
 static ucs_status_t
 ucp_mem_rcache_create(ucp_context_h context, const char *name,
-                      ucs_rcache_t **rcache_p, const ucs_rcache_config_t *config)
+                      ucs_rcache_t **rcache_p, ucs_rcache_params_t *rcache_params)
 {
-    ucs_rcache_params_t rcache_params;
-
-    ucs_rcache_set_params(config, &rcache_params);
-
-    rcache_params.region_struct_size = ucp_memh_size(context);
-    rcache_params.max_alignment      = ucs_get_page_size();
-    rcache_params.ucm_event_priority = 500; /* Default UCT pri - 1000 */
-    rcache_params.ucm_events         = UCM_EVENT_VM_UNMAPPED |
-                                       UCM_EVENT_MEM_TYPE_FREE;
-    rcache_params.context            = context;
-    rcache_params.ops                = &ucp_mem_rcache_ops;
-    rcache_params.flags              = UCS_RCACHE_FLAG_PURGE_ON_FORK;
-    rcache_params.alignment          = UCS_RCACHE_MIN_ALIGNMENT;
-
-    return ucs_rcache_create(&rcache_params, name, ucs_stats_get_root(),
+    rcache_params->region_struct_size = ucp_memh_size(context);
+    rcache_params->max_alignment      = ucs_get_page_size();
+    rcache_params->ucm_events         = UCM_EVENT_VM_UNMAPPED |
+                                        UCM_EVENT_MEM_TYPE_FREE;
+    rcache_params->context            = context;
+    rcache_params->ops                = &ucp_mem_rcache_ops;
+    rcache_params->alignment          = UCS_RCACHE_MIN_ALIGNMENT;
+
+    return ucs_rcache_create(rcache_params, name, ucs_stats_get_root(),
                              rcache_p);
 }
 
-ucs_status_t
-ucp_mem_rcache_init(ucp_context_h context, const ucs_rcache_config_t *cfg)
+ucs_status_t ucp_mem_rcache_init(ucp_context_h context,
+                                 const ucs_rcache_config_t *rcache_config)
 {
     ucs_status_t status;
+    ucs_rcache_params_t rcache_params;
 
-    status = ucp_mem_rcache_create(context, "ucp_rcache", &context->rcache, cfg);
+    ucs_rcache_set_default_params(&rcache_params);
+
+    rcache_params.alignment          = rcache_config->alignment;
+    rcache_params.ucm_event_priority = rcache_config->event_prio;
+    rcache_params.max_regions        = rcache_config->max_regions;
+    rcache_params.max_size           = rcache_config->max_size;
+    rcache_params.max_unreleased     = rcache_config->max_unreleased;
+    rcache_params.flags              = !rcache_config->purge_on_fork ? 0 :
+                                       UCS_RCACHE_FLAG_PURGE_ON_FORK;
```

Comment:
maybe add new function ucs_rcache_set_params() that will call ucs_rcache_set_default_params() and then initialize fields according to rcache_config. Then reuse it in UCT MD code as well.